added src and gitignore

manishmowall opened the pull request.

Commit:
added src and gitignore

Commit:
changes package name

naveen-webonise commented on src/main/java/org/webonise/concurrent_modification_exception/concurrent_modification_with_resolving/Application.java:
```diff
@@ -0,0 +1,31 @@
+package org.webonise.concurrent_modification_exception.concurrent_modification_with_resolving;
```
Wrong convention of package name. Please go through the standards followed to define package name

naveen-webonise commented on src/main/java/org/webonise/concurrent_modification_exception/concurrent_modification_with_resolving/Application.java:
```diff
@@ -0,0 +1,31 @@
+      list.add(4);
+      list.add(5);
+
+      Iterator<Integer> iterator = list.iterator();
```
Do we need to use `Iterator` for `CopyOnWriteArrayList`?

naveen-webonise commented on src/main/java/org/webonise/concurrent_modification_exception/concurrent_modification_with_resolving/Application.java:
```diff
@@ -0,0 +1,31 @@
+         System.out.println(value);
+
+         if (value.equals(3)) {
+            list.remove(2);
```
Why are you removing elements directly from `list` when you are using `Iterator`?

naveen-webonise replied:
Why 2 is hard coded?

naveen-webonise commented on src/main/java/org/webonise/concurrent_modification_exception/concurrent_modification_with_resolving/Application.java:
```diff
@@ -0,0 +1,31 @@
+
+      Iterator<Integer> iterator = list.iterator();
+
+      Integer value;
```
Why are you declaring `value` here? And why it's datatype is `Integer` ?

naveen-webonise commented on src/main/java/org/webonise/dining_philosophers/Application.java:
```diff
@@ -0,0 +1,29 @@
+public class Application {
+   public static void main(String[] args) {
+      //size - no of forks/philosophers
+      int size = 5;
```
This can be made constant

naveen-webonise commented on src/main/java/org/webonise/dining_philosophers/Philosopher.java:
```diff
@@ -0,0 +1,48 @@
+      try {
+         //to take thread out of loop and stop
+         int i = 0;
+         while (i < 5) {
```
Remove hard coded value

naveen-webonise replied:
Why do you want to loop till i reaches 5 here?

naveen-webonise commented on src/main/java/org/webonise/dining_philosophers/Application.java:
```diff
@@ -0,0 +1,29 @@
+         if (i < (size - 1)) {
+            new Philosopher(forks[i], forks[i + 1], i);
+         } else {
+            new Philosopher(forks[0], forks[i], i);
```
Why are you creating anonymous objects here?

naveen-webonise commented on src/main/java/org/webonise/dining_philosophers/Philosopher.java:
```diff
@@ -0,0 +1,48 @@
+      this.right = right;
+      this.id = "" + id;
+      thread = new Thread(this, this.id);
+      thread.start();
```
Dont start thread from constructor

naveen-webonise commented on src/main/java/org/webonise/producer_consumer/Application.java:
```diff
@@ -0,0 +1,30 @@
+      try {
+         producerThread.join();
+         consumerThread1.join();
+         consumerThread1.join();
```
Why is join() required here?

naveen-webonise commented on src/main/java/org/webonise/producer_consumer/Application.java:
```diff
@@ -0,0 +1,30 @@
+      List<Integer> sharedQueue = new ArrayList<>();
+
+      //this is the size of queue
+      final int size = 4;
```
This can be declared as  public static final in some file, as same value is being used in multiple places

naveen-webonise commented on src/main/java/org/webonise/producer_consumer/Consumer.java:
```diff
@@ -0,0 +1,41 @@
+
+      //Otherwise consume element and notify waiting producer
+      synchronized (sharedQueue) {
+         sharedQueue.notify();
```
Why are you calling notify() before consuming?

naveen-webonise reviewed (changes requested):
- Make changes wherever necessary as same mistakes are repeated in multiple places

Commit:
Concurrent Modification Exception Changes

1>changed package name
2>removed iterator and only using list
3>removed hard coded variable

Commit:
Dining Philosophers Changes
1> added constant variables
2> removed anonymous objects
3>removed start thread from constructor
4>removed hard coded variables

Commit:
Dining Philosophers Changes

1> added constant variables
2> removed anonymous objects
3> removed start thread from constructor
4> removed hard coded variables

Commit:
feature local HEAD behind remote HEAD
Merge branch 'feature' of https://github.com/nitishm-webonise/java-multithreading-assignments into feature

Commit:
Producer Consumer Changes

1> added TimeDelay class
2> removed hard coded variables
3> used static variables

manishmowall commented:
made changes that were asked
check it
please comment if required further corrections

Commit:
Dining Philosophers problem with interruption

Thread can be made stop by signally it to stop by using interrupt() and isInterrupted(). Rather then using our's own stop variable(runStatus).

Commit:
Producer Consumer Problem using Volatile Variable

using Volatile Keyword for runStatus variable to make sure that
changing the runStatus value is atomic action.

Commit:
Change name of getDeplay function in TimeDelay class of ProducerConsumer package
